graylog2-server/src/test/java/org/graylog/testing/elasticsearch/Client.java

```diff
@@ -211,4 +214,28 @@ public void assertSucceeded(JestResult jestResult) {
     public void refreshNode() {
         executeWithExpectedSuccess(new Refresh.Builder().build(), "Couldn't refresh elasticsearch node");
     }
+
+    public void bulkIndex(BulkIndexRequest bulkIndexRequest) {
+        final Bulk.Builder bulkBuilder = new Bulk.Builder().refresh(true);
+
+        bulkIndexRequest.requests()
+                .entrySet()
+                .stream()
+                .flatMap(entry -> {
+                            final String indexName = entry.getKey();
+
+                            return entry.getValue().stream()
+                                    .map(request -> new Index.Builder(request)
+                                            .index(indexName)
+                                            .type("test")
+                                            .refresh(true)
+                                            .build());
+                        })
+                .forEach(bulkBuilder::addAction);
+
+        final BulkResult bulkResult = executeWithExpectedSuccess(bulkBuilder.build(), "failed to execute bulk request");
+
+        assertThat(bulkResult.getFailedItems()).isEmpty();
+
+    }
```

Comment:
I would suggest to extract Lambda blocks to private methods:

    public void bulkIndex(BulkIndexRequest bulkIndexRequest) {
        final Bulk.Builder bulkBuilder = new Bulk.Builder().refresh(true);

        bulkIndexRequest.requests()
                .entrySet()
                .stream()
                .flatMap(this::indexActionsFromRequestEntries)
                .forEach(bulkBuilder::addAction);

        final BulkResult bulkResult = executeWithExpectedSuccess(bulkBuilder.build(), "failed to execute bulk request");

        assertThat(bulkResult.getFailedItems()).isEmpty();
    }

    private Stream<Index> indexActionsFromRequestEntries(Map.Entry<String, List<Map<String, Object>>> entry) {
        final String indexName = entry.getKey();

        return entry.getValue().stream()
                .map(request -> indexActionWithNameFromRequest(indexName, request));
    }

    private Index indexActionWithNameFromRequest(String indexName, Map<String, Object> request) {
        return new Index.Builder(request)
                .index(indexName)
                .type("test")
                .refresh(true)
                .build();
    }

Reply:
This will be done in a subsequent PR to avoid otherwise necessary rebasing and merging.